identity-backend/src/user/repository/customer.repository.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable } from "@nestjs/common";
+import { InjectModel } from "@nestjs/mongoose";
+import { Document, Model, Types } from "mongoose";
+import { IUserRepository } from "../interface/repository.interface";
+import { UserEntity } from "../entiry/user.entity";
+import { UserService } from "../service/user.service";
+import { User } from "./customer.model";
+
+@Injectable()
+export class UserRepository implements IUserRepository {
+  constructor(
+    @InjectModel('User') private userModel: Model<User>,
+    private userService: UserService,
+  ) {}
+
+  async getMe(id: string): Promise<UserEntity> {
+    let newUser;
+    
+    let exist = await this.userModel.findOne({ auth_id: id });
+    if (!exist) {
+      newUser = await this.userModel.create({ auth_id: id });
+    } else {
+      newUser = exist;
+    }
+
+    return newUser;
+  }
```

Comment:
Written a bit in complicated way. You use mutable variables, that is not as good as not mutable.
I guess could be better to use something like this:
`const user = await this.userModel.findOne({ auth_id: id });`
`if (!user) {`
`    return await this.userModel.create({ auth_id: id });`
`}`
`return user;`

Reply:
yes, at this stage your solution will be better. But let's say we don't want to show the user all the fields that we store in the database. Mappers are used for this.
I have updated the file